packages/lit/src/utils/range-manager.ts

```diff
@@ -177,185 +131,126 @@ export class RangeManager {
+        range.setEnd(endContainer, endOffset);
+      }
+      return range;
     }
-    const [inlineEditor, inlineRange] = result;
-
-    return inlineEditor.toDomRange(inlineRange);
   }
 
-  private _calculateInlineEditor(
-    point: TextRangePoint
-  ): [InlineEditor, InlineRange] | null {
-    const block = this.host.view.viewFromPath('block', point.path);
-    if (!block) {
-      return null;
-    }
-    const inlineEditorRoot = block.querySelector<InlineRootElement>(
-      `[${INLINE_ROOT_ATTR}]`
-    );
-    assertExists(
-      inlineEditorRoot,
-      `Cannot find inline editor element in block ${point.path.join(' > ')}}`
-    );
-
-    const maxLength = inlineEditorRoot.inlineEditor.yText.length;
-    const index = point.index >= maxLength ? maxLength : point.index;
-    const length =
-      index + point.length >= maxLength ? maxLength - index : point.length;
-
-    return [
-      inlineEditorRoot.inlineEditor,
-      {
-        index,
-        length,
-      },
-    ];
-  }
+  rangeToTextSelection(range: Range, reverse = false): TextSelection | null {
+    const { startContainer, endContainer } = range;
 
-  private _nodeToPoint(node: Node) {
-    const inlineEditorElement = this._getNearestInlineEditor(node);
-    if (!inlineEditorElement) {
+    const startBlock = this.getClosestBlock(startContainer);
+    const endBlock = this.getClosestBlock(endContainer);
+    if (!startBlock || !endBlock) {
       return null;
     }
-    const block = this._getBlock(inlineEditorElement);
-    if (!block) {
+
+    const startInlineEditor = this.getClosestInlineEditor(startContainer);
+    const endInlineEditor = this.getClosestInlineEditor(endContainer);
+    if (!startInlineEditor || !endInlineEditor) {
       return null;
     }
-    const inlineRange = this._range
-      ? inlineEditorElement.inlineEditor.toInlineRange(this._range)
-      : null;
-    if (!inlineRange) {
+
+    const startInlineRange = startInlineEditor.toInlineRange(range);
+    const endInlineRange = endInlineEditor.toInlineRange(range);
+    if (!startInlineRange || !endInlineRange) {
       return null;
     }
 
-    return {
-      blockId: block.model.id,
-      path: block.path,
-      index: inlineRange.index,
-      length: inlineRange.length,
-    };
-  }
-
-  private _snapshotToRange(snapshot: RangeSnapshot): Range {
-    const range = document.createRange();
-    range.setStart(snapshot.startContainer, snapshot.startOffset);
-    range.setEnd(snapshot.endContainer, snapshot.endOffset);
-    return range;
+    return this.host.selection.create('text', {
+      from: {
+        path: startBlock.path,
+        index: startInlineRange.index,
+        length: startInlineRange.length,
+      },
+      to:
+        startBlock === endBlock
+          ? null
+          : {
+              path: endBlock.path,
+              index: endInlineRange.index,
+              length: endInlineRange.length,
+            },
+      isReverse: reverse,
```

Comment:
Do we still need this property? I remember that this property was used to navigate with keyboard. Maybe we can try to remove it.

Reply:
Maybe we can use it to improve UX in remote-selection. 🤔

Reply:
You mean the position of the name flag?

Reply:
Yes

Reply:
And I want to rename `isReverse` to `reverse` 🤪